modules/sd_schedulers.py

```diff
@@ -31,13 +31,23 @@ def sgm_uniform(n, sigma_min, sigma_max, inner_model, device):
     return torch.FloatTensor(sigs).to(device)
 
 
+def kl_optimal(n, sigma_min, sigma_max, device):
+    alpha_min = torch.arctan(torch.tensor(sigma_min, device=device))
+    alpha_max = torch.arctan(torch.tensor(sigma_max, device=device))
+    sigmas = torch.empty((n+1,), device=device)
+    for i in range(n+1):
+        sigmas[i] = torch.tan((i/n) * alpha_min + (1.0-i/n) * alpha_max)
```

Comment:
It probably doesn't make much difference at any sane number of inference steps, but you can vectorize this operation if you want:
```suggestion
    step_indices = torch.arange(n + 1, device=device)
    sigmas = torch.tan(step_indices / n * alpha_min + (1.0 - step_indices / n) * alpha_max)
```
Also, I noticed that many schedulers [add a sigma of 0.0 at the end](https://github.com/comfyanonymous/ComfyUI/blob/master/comfy/k_diffusion/sampling.py#L22). I have no clue about schedulers in general, but is there a reason you didn't do that in `kl_optimal`?

Reply:
The code as written should end up with a zero at the end on its own, that's why it is over n+1.  Vectorized version looks like it maintains that part fine.